src/pydap/model.py

```diff
@@ -37,21 +55,54 @@
     >>> foo.dimensions = ('time',)
 
 Second, `BaseType` can hold data objects other than Numpy arrays. There are
-more complex data objects, like `pydap.proxy.ArrayProxy`, which acts as a
+more complex data objects, like `pydap.handlers.dap.BaseProxy`, which acts as a
 transparent proxy to a remote dataset, exposing it through the same interface.
 
 Now that we have some data, we can organize it using containers::
 
     >>> dataset = DatasetType('baz')
     >>> dataset['s'] = StructureType('s')
     >>> dataset['s']['foo'] = foo
+    >>> dataset['s']['bar'] = bar
+    >>> dataset['s']['foobar'] = foobar
 
 `StructureType` and `DatasetType` are very similar; the only difference is that
 `DatasetType` should be used as the root container for a dataset. They behave
 like ordered Python dictionaries::
 
-    >>> print(dataset.s.keys())
-    ['foo']
+    >>> print(list(dataset.s.keys()))
+    ['foo', 'bar', 'foobar']
+
+Slicing these datasets with a list of keywords yields a `StructureType`
+or `DatasetType` with only a subset of the children::
+
+    >>> print(dataset.s['foo', 'foobar'])
+    <StructureType with children 'foo', 'foobar'>
+    >>> print(list(dataset.s['foo', 'foobar'].keys()))
+    ['foo', 'foobar']
+
+In the same way, the ``.items()`` and ``.values()`` methods are like in python
+dictionnaries and they iterate over sliced values.
+
+The `StructureType`` never forgets of its children when sliced. Slicing a
+`StructureType` in fact simply changes its ``visible_keys`` property::
+
+    >>> print(dataset.s['foo', 'foobar'].visible_keys)
```

Comment:
This feels like a complex implementation detail that doesn't need to be exposed to users. What's the use for pulling out `visible_keys` rather than just looking at the dataset? Why use `all_keys()` or `all_values()` instead of keeping around the original object?